lib/trino-filesystem-manager/src/main/java/io/trino/filesystem/manager/FileSystemModule.java

```diff
@@ -91,6 +94,8 @@ protected void setup(Binder binder)
             install(new GcsFileSystemModule());
             factories.addBinding("gs").to(GcsFileSystemFactory.class);
         }
+
+        newOptionalBinder(binder, CachingHostAddressProvider.class).setDefault().to(NoneCachingHostAddressProvider.class).in(Scopes.SINGLETON);
```

Comment:
Rather than using an optional binder here, the configured cache implementation should bind the implementation. So the `NONE` cache should install `NoCachingHostAddressProvider` and `ALLUXIO` should install `ConsistentHashingHostAddressProvider`.